modules/virt-configuring-downward-metrics.adoc

```diff
@@ -0,0 +1,57 @@
+// Module included in the following assemblies:
+//
+// * virt/monitoring/virt-exposing-downward-metrics.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-configuring-downward-metrics_virt-using-downward-metrics"]
+= Configuring downward metrics
+
+You enable the capturing of downward metrics for a host VM by creating a configuration file that includes the following options:
+
+* A `downwardMetrics` device
+* A `virtio-serial` port, through which the metrics are exposed
+
+You must also enable a `DownwardMetrics` feature gate.
+
+.Prerequisites
+* Administrator privileges are required to enable the feature gate.
+
+.Procedure
```

Comment:
I would suggest to first enable the `donwardMetrics` feature gate with:

```bash
$ oc patch hco kubevirt-hyperconverged -n openshift-cnv \
  --type json -p '[{"op": "replace", "path": \
  "/spec/featureGates/downwardMetrics" \
  "value": true}]'
```

Reply:
yes looks better but i see there is a standard way followed for all featuregate. 
https://docs.openshift.com/container-platform/4.15/nodes/clusters/nodes-cluster-enabling-features.html

Reply:
I would suggest using this way : https://docs.openshift.com/container-platform/4.15/virt/vm_networking/virt-accessing-vm-secondary-network-fqdn.html . That's how we do it for deployKubeSecondaryDNS Feature Gate

Reply:
I'm not able to enable the `downwardMetrics` feature gate using either this method: https://docs.openshift.com/container-platform/4.15/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-enabling-features-console_nodes-cluster-enabling or this one: https://docs.openshift.com/container-platform/4.15/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-enabling-features-cli_nodes-cluster-enabling
As far as I was able to understand, both methods seem to be more intended for enabling more generic OpenShift features.

Reply:
I'm confused now. 

@jcanocan , are you suggesting that we remove the **Enabling the downward metrics feature gate** and **Disabling the downward metrics feature gate** topics/commands and show the feature gate configuration through a YAML file instead? I thought you said that wouldn't work.

Also, we say nothing earlier about the HyperConverged CR at all, so mentioning it here seems confusing.

Reply:
Sorry about the confusion. 

I'm not sure which option would be a better fit for this case.

I can find examples of toggling the feature gates using a [patch](https://github.com/openshift/openshift-docs/pull/74920#discussion_r1587242288) in the current documentation https://docs.openshift.com/container-platform/4.15/virt/storage/virt-automatic-bootsource-updates.html as well as editing the CR directly in an editor.

This leads me to think that there is not a standard way to enable and disable feature gates. IMHO, the [patch](https://github.com/openshift/openshift-docs/pull/74920#discussion_r1587242288) approach is more straight away and less error prune. WDYT @geetikakay?

In any case, maybe we can squash both sections into one (e.g., Managing downward metrics feature gate), indicating that a `true` value on the `spec.featureGates.downwardMetrics` will enable the FG while a `false` will disable it. Another option is to also squash both sections into one, as previously suggested, and show both patches commands, but following this idea: https://docs.openshift.com/container-platform/4.15/virt/storage/virt-automatic-bootsource-updates.html

AFAIK, [this approach](https://github.com/openshift/openshift-docs/pull/74920#discussion_r1594179450) will work without any issues. Please note that the yaml file we are suggesting to edit is the configuration of the deployed components in charge of managing OCP-Virt. Therefore, a change to this configuration file will cause the underlying infrastructure to reconcile and pick up the updated configuration, i.e., enabling/disabling the feature gate.

Regarding the `HyperConverged CR` term definition, other docs entries, e.g., [CPU models](https://docs.openshift.com/container-platform/4.15/virt/nodes/virt-managing-node-labeling-obsolete-cpu-models.html), [FQDN](https://docs.openshift.com/container-platform/4.15/virt/vm_networking/virt-accessing-vm-secondary-network-fqdn.html) or [boot sources](https://docs.openshift.com/container-platform/4.15/virt/storage/virt-automatic-bootsource-updates.html), does not define the term HyperConverged CR, at most the first occurrence of `CR` is defined as `custom resource`. IMHO, the doc entries assume an educated reader in the basic terms.

Reply:
Enabling Downward metrics feature gate

1. Enable the downward metrics. Open the HyperConverged CR in your default editor by running the following command
$ oc edit hyperconverged kubevirt-hyperconverged -n openshift-cnv


Step 2: Edit the HyperConverged CR, to enable feature gate 


apiVersion: hco.kubevirt.io/v1beta1
kind: HyperConverged
metadata:
  name: kubevirt-hyperconverged
  namespace: openshift-cnv
spec:
  featureGates:
    downwardMetrics: false
# ...

Alternatively, To enable the downwardMetrics feature gate using the command line interface (CLI), use the oc patch command as follows:

$ oc patch hco kubevirt-hyperconverged -n openshift-cnv \
  --type json -p '[{"op": "replace", "path": \
  "/spec/featureGates/downwardMetrics" \
  "value": true}]'


Disabling Downward metrics feature gate
...

Just a thought , probably like this @aspauldi @jcanocan .